node/container.go

```diff
@@ -722,3 +719,33 @@ func (a *HostAgent) createContainerConfig(tenantID string, svc *service.Service,
 	}
 	return cfg, hcfg, state, nil
 }
+
+func (a *HostAgent) createContainer(conf *dockerclient.Config, hostConf *dockerclient.HostConfig, svcID string, instanceID int) (*docker.Container, error) {
+	logger := plog.WithFields(log.Fields{
+		"serviceid":  svcID,
+		"instanceid": instanceID,
+	})
+
+	if hostConf == nil {
+		logger.Error("Host Config passed to createContainer is nil.")
```

Comment:
That's a good question. 

Looking at the code, I also have a redundant version of the same check about 9 lines later (obvious now that we've factored out CreateContainerOptions). 

In the code as it is currently written, there is only one path to createContainer, and it ensures that  hostConf will be non-nil.  So, that's an argument for removing the check altogether. 

On the other hand, if future code should call the function with a nil hostConf, we'll get a panic here:  https://github.com/control-center/serviced/blob/develop/commons/docker/api.go#L108
That's an argument for leaving the check in and making it an error (and possibly returning an error). 

I guess my preference would be eliminate the redundant check, and make the first one an error (returning an error). 

I am not particularly opinionated about this, so if you have a different preference, I'll be glad to implement that.